consumer/consumererror/internal/statusconversion/conversion_test.go

```diff
@@ -0,0 +1,113 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package statusconversion // import "go.opentelemetry.io/collector/consumer/consumererror/internal/statusconversion"
+
+import (
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func Test_GetHTTPStatusCodeFromStatus(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    *status.Status
+		expected int
+	}{
+		{
+			name:     "Retryable Status",
+			input:    status.New(codes.Unavailable, "test"),
+			expected: http.StatusServiceUnavailable,
+		},
+		{
+			name:     "Non-retryable Status",
+			input:    status.New(codes.InvalidArgument, "test"),
+			expected: http.StatusInternalServerError,
+		},
+		{
+			name:     "Specifically 429",
+			input:    status.New(codes.ResourceExhausted, "test"),
+			expected: http.StatusTooManyRequests,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := GetHTTPStatusCodeFromStatus(tt.input)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+}
+
+func Test_ErrorMsgAndHTTPCodeToStatus(t *testing.T) {
+	tests := []struct {
+		name       string
+		errMsg     string
+		statusCode int
+		expected   *status.Status
+	}{
+		{
+			name:       "Bad Request",
+			errMsg:     "test",
+			statusCode: http.StatusBadRequest,
+			expected:   status.New(codes.InvalidArgument, "test"),
+		},
+		{
+			name:       "Unauthorized",
+			errMsg:     "test",
+			statusCode: http.StatusUnauthorized,
+			expected:   status.New(codes.Unauthenticated, "test"),
+		},
+		{
+			name:       "Forbidden",
+			errMsg:     "test",
+			statusCode: http.StatusForbidden,
+			expected:   status.New(codes.PermissionDenied, "test"),
+		},
+		{
+			name:       "Not Found",
+			errMsg:     "test",
+			statusCode: http.StatusNotFound,
+			expected:   status.New(codes.Unimplemented, "test"),
```

Comment:
Any reason that this doesn't map to `codes.NotFound (5)`. [This](https://github.com/googleapis/googleapis/blob/d52dddfd3c0eee864228772c765ad817bba15f6f/google/rpc/code.proto#L69-L78) seems to indicate that would a valid mapping.

Reply:
That's a good catch. In fact, we should see how much of this file we can adopt since these seem to be as close to canonical mappings as are available.

I'd like to make this change later since this logic is supposed to be a straight copy of what we have in the OTLP receiver, but I'm open to making it in this PR if others feel differently. cc @TylerHelmuth

Reply:
I'd prefer doing it later. The current mappings are based on https://github.com/grpc/grpc/blob/master/doc/http-grpc-status-mapping.md, which to my knowledge was the source of truth for mappings.  If those are incorrect, I'd prefer to handle it in a separate issue/PR.